fitlins/interfaces/bids.py

```diff
@@ -161,8 +161,20 @@ class LoadBIDSModel(SimpleInterface):
     def _run_interface(self, runtime):
         from bids.analysis import Analysis
         from bids.layout import BIDSLayout
+        import re
+
         force_index = self.inputs.force_index or None
-        ignore = self.inputs.ignore or None
+        if self.inputs.ignore:
+            ignore = []
+            for patt in self.inputs.ignore:
+                # If starts and ends with `/` `/`, treat as aregex
+                if patt[0] == '/' and patt[-1] == '/':
+                    patt = re.compile(patt[1:-1])
+                ignore.append(patt)
+        else:
+            ignore = None
```

Comment:
Just to be a little more concise, the whole `if/else` block can be a list comprehension:

```Python
ignore = [
    # If entry looks like `/<pattern>/`, treat `<pattern>` as a regex
    re.compile(ign[1:-1]) if (ign[0], ign[-1]) == ('/', '/') else ign
    # Iterate over empty tuple if undefined
    for ign in self.inputs.ignore or ()]
```

Reply:
I considered that but didn't want to be too Rubyesque. But the comments help a lot.

Reply:
Yeah, each line doesn't take much work to understand, but the two together are pretty dense without the comments. But I loathe `for: append()` loops.

Reply:
Same. I think yours is better though so I'll add it.